src/core/Hamiltonian.jl

```diff
@@ -55,6 +55,7 @@ Update Hamiltonian in-place
 function update_hamiltonian!(ham::Hamiltonian, ρ::Density)
     basis = ham.basis
     model = basis.model
+    ham.density = ρ
```

Comment:
Don't we better want to in-place update the real and fourier part of the density. I don't fully remember, but it might be that it is assumed somewhere that the pointer addresses will not change. Did you check that?

Reply:
No, but it looked relatively harmless so I just changed the type to be mutable. We could do in-place update here. What's the way to do it properly? Should we overload `copy!`?

Reply:
Good question ... I never looked at that. Overloading `copy!` sounds reasonable. Can one also overload the `.=` operator?

Reply:
`methods(copy!)` has a few results so at least it's not explicitly discouraged. Not sure if it matters, but I pushed it.